app/src/main/java/com/zarinraim/dogsearch/presentation/main/all_breeds/DogBreedsListModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.zarinraim.dogsearch.presentation.main.all_breeds
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.zarinraim.dogsearch.data.model.toAllBreeds
+import com.zarinraim.dogsearch.domain.model.AllBreeds
+import com.zarinraim.dogsearch.domain.repository.BreedsRepository
+import kotlinx.coroutines.launch
+
+class DogBreedsListModel(private val repo: BreedsRepository) : ViewModel() {
+    val dogBreeds: MutableState<AllBreeds> = mutableStateOf(AllBreeds(mapOf()))
```

Comment:
The common practice is not to propagate anything mutable outside a view model. Probably you should use two properties for this case: just public `State`, which is visible from the compose Screen, and private mutable State.

Reply:
Done.